app/upgrades.go

```diff
@@ -9,21 +9,105 @@ import (
 )
 
 func (app *ChainApp) RegisterUpgradeHandlers(cdc codec.BinaryCodec) {
-	planName := "v5.0"
+	planName := "v6.0.0"
+
+	// Set param key table for params module migration
+	for _, subspace := range app.ParamsKeeper.GetSubspaces() {
+		var keyTable paramstypes.KeyTable
+		switch subspace.Name() {
+		case authtypes.ModuleName:
+			keyTable = authtypes.ParamKeyTable() //nolint:staticcheck
+		case banktypes.ModuleName:
+			keyTable = banktypes.ParamKeyTable() //nolint:staticcheck
+		case stakingtypes.ModuleName:
+			keyTable = stakingtypes.ParamKeyTable() //nolint:staticcheck
+		case minttypes.ModuleName:
+			keyTable = minttypes.ParamKeyTable() //nolint:staticcheck
+		case distrtypes.ModuleName:
+			keyTable = distrtypes.ParamKeyTable() //nolint:staticcheck
+		case slashingtypes.ModuleName:
+			keyTable = slashingtypes.ParamKeyTable() //nolint:staticcheck
+		case govtypes.ModuleName:
+			keyTable = govv1.ParamKeyTable() //nolint:staticcheck
+		case ibcexported.ModuleName:
+			keyTable = ibcclienttypes.ParamKeyTable()
+			keyTable.RegisterParamSet(&ibcconnectiontypes.Params{})
+		case ibctransfertypes.ModuleName:
+			keyTable = ibctransfertypes.ParamKeyTable()
+		case icacontrollertypes.SubModuleName:
+			keyTable = icacontrollertypes.ParamKeyTable()
+		case icahosttypes.SubModuleName:
+			keyTable = icahosttypes.ParamKeyTable()
+		// case icaauthmoduletypes.ModuleName:
+		// 	keyTable = icaauthmoduletypes.ParamKeyTable()
+		default:
+			continue
+		}
+		if !subspace.HasKeyTable() {
+			subspace.WithKeyTable(keyTable)
+		}
+	}
+	baseAppLegacySS := app.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
+
 	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+		// OPTIONAL: prune expired tendermint consensus states to save storage space
+		if _, err := ibctmmigrations.PruneExpiredConsensusStates(sdkCtx, cdc, app.IBCKeeper.ClientKeeper); err != nil {
+			return nil, err
+		}
+		// explicitly update the IBC 02-client params, adding the localhost client type
+		var allowedClients []string
+		ibcParamSpace := app.GetSubspace(ibcexported.ModuleName)
+		ibcParamSpace.Get(sdkCtx, ibcclienttypes.KeyAllowedClients, &allowedClients)
+		allowedClients = append(allowedClients, ibcexported.Localhost)
+		ibcParamSpace.SetParamSet(sdkCtx, &ibcclienttypes.Params{
+			AllowedClients: allowedClients,
+		})
+		if err := ibcmigrationsv6.MigrateICS27ChannelCapability(
+			sdkCtx,
+			cdc,
+			app.keys["capability"],
+			app.CapabilityKeeper,
+			icacontrollertypes.SubModuleName,
+		); err != nil {
+			return nil, err
+		}
+
+		// Migrate Tendermint consensus parameters from x/params module to a dedicated x/consensus module.
+		baseapp.MigrateParams(sdkCtx, baseAppLegacySS, &app.ConsensusParamsKeeper.ParamsStore)
```

Comment:
same as https://github.com/crypto-org-chain/chain-main/pull/1155, but just miss to handle err